.vsts.pipelines/jobs/collect-artifacts.yml

```diff
@@ -0,0 +1,104 @@
+jobs:
+- job: Collect_Artifacts
+
+  dependsOn: ${{ parameters.dependsOn }}
+
+  displayName: Collect Artifacts
+
+  pool: ${{ parameters.pool }}
+
+  steps:
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - task: DownloadBuildArtifacts@0
+      displayName: Download CentOS artifacts
+      inputs:
+        artifactName: Tarball centos71 Offline
+        downloadPath: '$(Build.StagingDirectory)/centos'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Debian artifacts
+      inputs:
+        artifactName: Tarball debian9 Offline
+        downloadPath: '$(Build.StagingDirectory)/debian'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Fedora artifacts
+      inputs:
+        artifactName: Tarball fedora29 Offline
+        downloadPath: '$(Build.StagingDirectory)/fedora'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Ubuntu artifacts
+      inputs:
+        artifactName: Tarball ubuntu1604 Offline
+        downloadPath: '$(Build.StagingDirectory)/ubuntu'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download portable artifacts
+      inputs:
+        artifactName: Tarball centos71 Offline Portable
+        downloadPath: '$(Build.StagingDirectory)/portable'
+
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        workingDirectory: '$(Build.StagingDirectory)'
+        script: |
+          version=`echo centos/*/Private.SourceBuilt.Artifacts*.tar.gz | sed 's/\.tar\.gz$//' | sed 's/^centos\/.*\/Private\.SourceBuilt\.Artifacts\.//'`
```

Comment:
`$(` is preferred over <code>`</code> generally. But this is also quite complex to be here and something that might belong as build logic (where I would *think* version would be accessible?).